Packaging: Add files to create tendrl-ceph-bridge rpm

TimothyAsir opened the pull request:
Signed-off-by: Timothy Asir J tjeyasin@redhat.com#TEN-38

TimothyAsirJeyasing committed:
Packaging: Add files to create tendrl-ceph-bridge rpm

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing committed:
Packaging: Add files to create tendrl-ceph-bridge rpm

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

shtripat commented on Makefile:
```diff
@@ -0,0 +1,39 @@
+	if [ "$$EC" -eq "0" ]; then \
+		FILE=$$(readlink -f $$(find $(RPMBUILD)/RPMS -name tendrl-ceph-bridge-$(VERSION)*.rpm)); \
+		cp -f $$FILE $(DEPLOY)/latest/; \
+		printf "\nThe bridge common RPMs are located at:\n\n"; \
```
Is this message for bridge_common or it should be ceph_bridge?

TimothyAsir replied:
Yes it should be ceph_bridge only.
I will correct this

shtripat commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+Summary: Tendrl bridge for Ceph Storage
+Source0: %{pkg_name}-%{pkg_version}.tar.gz
+Group:   Applications/System
+License: GPLv2+
```
License is LGPL I remember

TimothyAsir replied:
Sure i will change to LGPL2.1

shtripat commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+rm -rf {test-,}requirements.txt
+
+# Remove bundled egg-info
+rm -rf %{pkg_name}.egg-info
```
The .egg-info file shouldn't be there in TAR ball itself I feel.

TimothyAsir replied:
Sure, i will remove in tar also.
Keeping it here will help if someone uses rpmbuild command.

shtripat commented on tendrl-cephd.service:
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=Tendrl Ceph Daemon to Manage ceph tasks
```
Description should be more meaningful. May be like "Tendrl ceph daemon to manage ceph storage system"

TimothyAsir replied:
Sure

TimothyAsirJeyasing committed:
Packaging: Updated licence and some minor details

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing committed:
Packaging: Updated licence and some minor details

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsir commented:
@r0h4n, @shtripat, @nthomas-redhat, @ktderyer,
I have updated the patch.  Please review

ktdreyer commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+
+Name: %{pkg_name}
+Version: %{pkg_version}
+Release: %{pkg_release}%{?dist}
```
This boilerplate duplication can be simplified:

```
Name: tendrl-ceph-bridge
Version: 0.0.1
Release: 1%{?dist}
```

And then use `%{name}` and `%{version}` macros throughout the packaging, instead of the non-standard `%{pkg_name}` or `%{pkg_version}`

TimothyAsir replied:
Done

ktdreyer commented on Makefile:
```diff
@@ -0,0 +1,40 @@
+CWD := $(shell pwd)
+BASEDIR := $(CWD)
+PRINT_STATUS = export EC=$$?; cd $(CWD); if [ "$$EC" -eq "0" ]; then printf "SUCCESS!\n"; else exit $$EC; fi
+VERSION=0.0.1
```
This version number does not match `setup.py` (`0.0.1` versus `0.1`)

TimothyAsir replied:
Done

ktdreyer commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+BuildArch: noarch
+Summary: Tendrl bridge for Ceph Storage
+Source0: %{pkg_name}-%{pkg_version}.tar.gz
+Group:   Applications/System
```
No need for `Group:`. I recommend you drop this line.

TimothyAsir replied:
Done

ktdreyer commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+Summary: Tendrl bridge for Ceph Storage
+Source0: %{pkg_name}-%{pkg_version}.tar.gz
+Group:   Applications/System
+License: LGPL2.1
```
This `License:` field should be one of the values from https://fedoraproject.org/wiki/Licensing:Main

For example, "License: LGPLv2"

TimothyAsir replied:
Sure, I will change to LGPLv2+

ktdreyer commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+Source0: %{pkg_name}-%{pkg_version}.tar.gz
+Group:   Applications/System
+License: LGPL2.1
+Url: https://github.com/Tendrl/ceph_bridge
```
Url -> URL

TimothyAsir replied:
Done

ktdreyer commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+Requires: pytz
+Requires: python-msgpack >= 0.4.5
+Requires: python-taskflow >= 2.6.0
+Requires: tendrl-bridge-common
```
Please add `Requires: systemd` as well.

There are no `BuildRequires:` lines here. I think you'll need (at a minimum):

```
BuildRequires: systemd
BuildRequires: python2-devel
BuildRequires: python-sphinx
```

... but you'll need to run this through the "mock" build tool to be certain. Does this package build in mock for EL7? Try rebuilding your SRPM to find out:

```
mock -r epel-7-x86_64 rebuild tendrl-ceph-bridge-0.0.1-1.el7.src.rpm
```

(In fact it would probably be safer if the Makefile used `mock` instead of calling `rpmbuild` directly.)

ktdreyer commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+
+%install
+%{__python} setup.py install --single-version-externally-managed -O1 --root=$RPM_BUILD_ROOT --record=INSTALLED_FILES
+install -Dm 0644 tendrl-cephd.service $RPM_BUILD_ROOT/usr/lib/systemd/system/tendrl-cephd.service
```
Please use the systemd macro `%{_unitdir}` instead of hard-coding `/usr/lib/systemd/system` here.

(See `rpm --eval %{_unitdir}` to view its value.)

TimothyAsir replied:
Done

ktdreyer commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+%files -f INSTALLED_FILES
+%doc html README.rst
+%license LICENSE
+%{_usr}/lib/systemd/system/tendrl-cephd.service
```
Please use the systemd macro `%{_unitdir}` instead of hard-coding `%{_usr}/lib/systemd/system` here.

TimothyAsir replied:
Done

ktdreyer commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+%install
+%{__python} setup.py install --single-version-externally-managed -O1 --root=$RPM_BUILD_ROOT --record=INSTALLED_FILES
+install -Dm 0644 tendrl-cephd.service $RPM_BUILD_ROOT/usr/lib/systemd/system/tendrl-cephd.service
+
```
Please run the test suite during the RPM build to ensure that the tests pass for this specific OS platform and Python version. You'll need to add `BuildRequires: pytest`, and then add a `%check` section like so:

```
%check
py.test -v tendrl/ceph_bridge/tests
```

TimothyAsir replied:
Done

ktdreyer commented on ceph_bridge.spec:
```diff
@@ -0,0 +1,68 @@
+Tendrl bridge for Ceph Storage
+
+%prep
+%setup -n %{pkg_name}-%{pkg_version}
```
No need for `-n <anything>` here... This is already %setup's default.

TimothyAsir replied:
Done

ktdreyer commented on tendrl-cephd.service:
```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=
```
`Description=` is blank here

TimothyAsir replied:
Done

mbukatov commented on Makefile:
```diff
@@ -0,0 +1,40 @@
+	mkdir -p $(RPMBUILD)/SPECS
+	sed -e "s/@VERSION@/$(VERSION)/" ceph_bridge.spec \
+	        > $(RPMBUILD)/SPECS/ceph_bridge.spec
+	rpmbuild -ba $(RPMBUILD)/SPECS/ceph_bridge.spec
```
As Ken already points out, it would be good to build the package via mock (compared to just `rpmbuild -ba`), so that one can be sure that the package is build in a clean environment.

ktdreyer replied:
Please see https://github.com/ktdreyer/errata-tool/blob/master/Makefile for an example `Makefile` that determines the Python package version dynamically and simplifies the build operations with fedpkg and mock.

TimothyAsir replied:
Thank you for pointing me the reference document which really helped me a lot.
I will update the make file to use mock instead of rpmbuild.

TimothyAsirJeyasing committed:
Packaging: Enhance rpm spec file and make file

Updated make file to package via mock
Fixed tendrl-cephd.service description issue
Corrected package name and version details in setup.py
Used systemd macro instead of hard-coded paths
Added pytest to run the test suite during the rpm build
Added build requires packages
Updated the licensing information

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing committed:
Packaging: Enhance rpm spec file and make file

Updated make file to package via mock
Fixed tendrl-cephd.service description issue
Corrected package name and version details in setup.py
Used systemd macro instead of hard-coded paths
Added pytest to run the test suite during the rpm build
Added build requires packages
Updated the licensing information

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsir commented on tendrl-cephd.service:
```diff
@@ -1,14 +0,0 @@
-[Unit]
```
This file is moved to systemd directory. I forget to add it here. I will add in the next patch.

TimothyAsir replied:
Done

TimothyAsirJeyasing committed:
Packaging: Added tendrl-cephd.service file which was missing

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing committed:
Packaging: Updated the build file as per directory changes

1) Updated spec file and makefile and setup.py file
2) spec file name changed to tendrl-ceph-integration.spec

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing committed:
Merge branch 'packaging' of https://github.com/TimothyAsir/ceph_bridge into packaging

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

r0h4n committed:
Merge branch 'master' into packaging
